modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/BinaryUtils.cs

```diff
@@ -642,14 +650,175 @@ public static unsafe void WriteString(string val, IBinaryStream stream)
 
             fixed (char* chars = val)
             {
-                int byteCnt = Utf8.GetByteCount(chars, charCnt);
+                int byteCnt = GetUtf8ByteCount(chars, charCnt);
 
                 stream.WriteInt(byteCnt);
 
                 stream.WriteString(chars, charCnt, byteCnt, Utf8);
             }
         }
 
+        /// <summary>
+        /// Converts string to UTF8 bytes.
+        /// </summary>
+        /// <param name="chars">Chars.</param>
+        /// <param name="charCnt">Chars count.</param>
+        /// <param name="byteCnt">Bytes count.</param>
+        /// <param name="enc">Encoding.</param>
+        /// <param name="data">Data.</param>
+        /// <returns>Amount of bytes written.</returns>
+        public static unsafe int StringToUtf8Bytes(char* chars, int charCnt, int byteCnt, Encoding enc, byte* data)
+        {
+            if (!UseStringSerializationVer2)
+                return enc.GetBytes(chars, charCnt, data, byteCnt);
+
+            int strLen = charCnt;
+            int c, cnt;
+
+            int position = 0;
+
+            for (cnt = 0; cnt < strLen; cnt++)
+            {
+                c = *(chars + cnt);
+
+                if (c <= 0x007F)
```

Comment:
Pavel,

I changed this condition on Java side from (c <= 0x007F) to if (c >= 0x0001 && c <= 0x007F). Please update on .Net side.